### PR TITLE
Allow uppercasing change ids

### PR DESCRIPTION
**Pull request by ilyagr**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- (n/a) I have updated the config schema (src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: src/template_parser.rs [hunk 1]
```diff
@@ -906,6 +906,20 @@ fn build_string_method<'a, L: TemplateLanguage<'a>>(
                 TemplatePropertyFn(|s: &String| s.lines().next().unwrap_or_default().to_string()),
             ))
         }
+        "upper" => {
```
Comment (samueltardieu) [inline src/template_parser.rs:909]: nit: at some point we should start using `const &'static str`, we have a risk of forgetting to update a string literal potentially used at several places.

File: tests/test_commit_template.rs [hunk 2]
```diff
@@ -180,12 +180,32 @@ fn test_log_customize_short_id() {
...
+       (empty) (no description set)
+    "###);
+
+    // Strings like those returned by `short()` can be uppercased
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "log",
+            "--config-toml",
+            &format!("{decl}='id.short().upper()'"),
```
Comment (samueltardieu) [inline tests/test_commit_template.rs:202]: Shouldn't `.lower()` be tested as well (for example on a literal)? Also, isn't that lacking a test for `.shortest().upper()` (and `.upper().lower()`)?

Comment (yuja) [reply tests/test_commit_template.rs:202]: Perhaps, using `-T template` is easier to test various combinations.

For string methods, we can also add a test in template_parser.rs, which should be cheaper than spawning jj process.

File: src/config/templates.toml [hunk 3]
```diff
@@ -47,6 +47,8 @@ show = 'show'
 
 # Hook points for users to customize the default templates:
 'format_short_id(id)' = 'id.shortest(12)'
+'format_change_id(id)' = 'format_short_id(id)'
+'format_commit_id(id)' = 'format_short_id(id)'
```
Comment (yuja) [inline src/config/templates.toml:51]: nit: `format_short_commit_id()`/`_change_id()` because `jj show` doesn't use these for formatting full id.

**Review by yuja — APPROVED** [review]

File: src/template_parser.rs (hunk 1 above)
Comment (ilyagr) [reply src/template_parser.rs:909]: I think it's OK to mostly relate on tests for this. Moreover, I don't think this particular string is repeated in the code (as opposed to tests or docs) in a way where it has to match exactly.

File: src/commit_templater.rs [hunk 4]
```diff
@@ -382,6 +382,21 @@ impl Template<()> for ShortestIdPrefix {
     }
 }
 
+impl ShortestIdPrefix {
```
[the remark this answers is not in the document]
Comment (ilyagr) [reply src/commit_templater.rs:385]: No problem.

File: tests/test_commit_template.rs (hunk 2 above)
Comment (ilyagr) [reply tests/test_commit_template.rs:202]: I agree, at least on of the tests I added is extraneous. I'll try to move it somewhere, maybe in a day or two.

Comment (ilyagr) [reply tests/test_commit_template.rs:202]: I think I'll just add a test to `test_templater.rs`. It does spawn a lot of `jj` processes, but the tests fit in and if we want to call `jj` less we'd need to rewrite those tests anyway.

**Review by yuja — APPROVED** [review]
LGTM, thanks.

**Comment by ilyagr** [comment]
Thanks for the review!

**Pull request merged**